luaui/Widgets/cmd_blueprint.lua

```diff
@@ -267,7 +267,7 @@ local function setSelectedBlueprintIndex(index)
 		WG["api_blueprint"].setActiveBlueprint(nil)
 	end
 
-	if blueprintPlacementActive and index ~= nil and index > 0 then
+	if index ~= nil and index > 0 then
 		Spring.Echo("[Blueprint] selected blueprint #" .. selectedBlueprintIndex)
 	end
```

Comment:
In the case where blueprint placement is active, it's a bit redundant. If it's not (selecting while inactive is a new case this PR introduces), then it's the only indication that something happened. I could imagine removing a lot of the `Spring.Echo`s; I do feel like actions should give some feedback to the player though. I'm not sure what the best way to do that is.

Reply:
You can make it not echo at least when the blueprint gui is open. It's a good compromise imo, since some feedback when nothing visible is seen is good.

Reply:
Keep in mind echos are for dev work, not for anything player-facing.